Objects/abstract.c

```diff
@@ -1859,8 +1859,16 @@ PySequence_Tuple(PyObject *v)
     if (n == -1)
         goto Fail;
     result = PyTuple_New(n);
-    if (result == NULL)
-        goto Fail;
+    if (result == NULL) {
+        /* bpo-28940 - LengthHint could lie and asked for too much,
+         * try again without preallocation. */
+        PyErr_Clear();
```

Comment:
This will silence any possible exception raised in `PyTuple_New()` (same for `PyByteArray_FromStringAndSize()`, `_PyBytesWriter_Alloc()`, etc.). I don't think that's the right approach.